crates/dc_bundle/src/definition.rs

```diff
@@ -22,3 +25,132 @@ pub mod plugin;
 pub mod view;
 
 include!(concat!(env!("OUT_DIR"), "/designcompose.definition.rs"));
+
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Hash, Serialize, Deserialize)]
+pub enum NodeQuery {
+    /// Find the node by ID
+    NodeId(String),
+    /// Find the node by name
+    NodeName(String),
+    /// Node by name that is a variant, so the name may have multiple properties
+    /// The first string is the node name and the second is its parent's name
+    NodeVariant(String, String),
+    NodeComponentSet(String),
+}
+
+impl NodeQuery {
+    /// Construct a NodeQuery::NodeId from the given ID string.
+    pub fn id(id: impl ToString) -> NodeQuery {
+        NodeQuery::NodeId(id.to_string())
+    }
+
+    /// Construct a NodeQuery::NodeName from the given ID string.
```

Comment:
I know these comments are copied but some are wrong. Can you fix them? This should be "from the given node name"